R/drop_search.R

```diff
@@ -1,46 +1,57 @@
 
 
+
 #'Returns metadata for all files and folders whose filename contains the given
 #'search string as a substring.
 #'
 #'@param query  The search string. This string is split (on spaces) into
 #'  individual words. Files and folders will be returned if they contain all
 #'  words in the search string.
 #'@template path
-#'@template file_limit
-#'@param  include_deleted If this parameter is set to true, then files and
-#'  folders that have been deleted will also be included in the search.
-#'@template locale
-#'@template include_membership
+#'@param start The starting index within the search results (used for paging).
+#'  The default for this field is 0
+#'@param max_results The maximum number of search results to return. The default
+#'  for this field is 100.
+#'@param  mode Mode can take the option of filename, filename_and_content, or search deleted files with deleted_filename
 #'@template token
 #'@export
 #' @examples \dontrun{
 #' # If you know me, you know why this query exists
 #' drop_search('gif') %>% select(path, is_dir, mime_type)
 #'}
 drop_search <- function(query = NULL,
-                        path = NULL,
-                        file_limit = 1000,
-                        include_deleted = FALSE,
-                        locale = NULL,
-                        include_membership = FALSE,
+                        path = "",
+                        start = 0,
+                        max_results = 100,
+                        mode = "filename",
                         dtoken = get_dropbox_token()) {
-    args <- as.list(drop_compact(c(query = query,
-                                 path = path,
-                                 file_limit = file_limit,
-                                 include_deleted = include_deleted,
-                                 locale = locale,
-                                 include_membership = include_membership)))
-    search_url <- "https://api.dropbox.com/1/search/auto/"
-    res <- GET(search_url, query = args, config(token = dtoken))
-    results <- content(res)
-    # The search results contain a field called 'modifier' that is NULL
-    # by default. This makes it really hard to rbind the list either with
-    # dplyr or data.table. So this lapply just kills that field before I
-    # can rbind everything into a nice list.
-    zz <- lapply(results, function(t) { t$modifier = NULL; t })
-    # I hate myself :(
-    dplyr::tbl_df(data.table::rbindlist(zz, fill = T))
-}
+  available_modes <-
+    c("filename", "filename_and_content", "deleted_filename")
+  assertive::assert_any_are_matching_fixed(available_modes, mode)
 
 
+  args <- as.list(drop_compact(
+    c(
+      query = query,
+      path = path,
+      start = as.integer(start),
+      max_results = as.integer(max_results),
+      mode = mode
+    )
+  ))
+  args$start <- as.integer(start)
```

Comment:
these two `args$foobar <- ` lines look unnecessary

Reply:
Without these, the integers become coerced into characters and the API throws an error.

Reply:
That's because of the `c` inside `drop_compact`, neither of which is really necessary; none of the arguments should ever be null, right?

What if it was just
```r
args <- list(
  query = query,
  path = path,
  start = as.integer(start),
  max_results = as.integer(max_results),
  mode = mode
)
```

If you wanted to keep the `drop_compact`, it should be `drop_compact(list(...))`, not `as.list(drop_compact(c(...)))`.

Reply:
Thanks! I totally missed that.

Reply:
Fixed in `8b05b88`